File: exercises/day14-Santa-package-factory-exercises/csharp/PackagingService/Child.cs
namespace PackagingService
{
    public class Child
    {
        public string Name { get; }
        public int Age { get; }
        public ChildGender Gender { get; }
        public bool HasBeenNice { get; }
        public Gift AssignedGift { get; }
        
        public Child(string name, int age, ChildGender gender, 
                     bool hasBeenNice, Gift assignedGift)
        {
            Name = name;
            Age = age;
            Gender = gender;
            HasBeenNice = hasBeenNice;
            AssignedGift = assignedGift;
        }
    }
}

File: exercises/day14-Santa-package-factory-exercises/java/src/main/java/packagingservice/Child.java
package packagingservice;

public class Child {

    private final String name;
    private final int age;
    private final ChildGender gender;
    private final boolean hasBeenNice;
    private final Gift assignedGift;

    public Child(String name, int age, ChildGender gender,
                 boolean hasBeenNice, Gift assignedGift) {
        this.name = name;
        this.age = age;
        this.gender = gender;
        this.hasBeenNice = hasBeenNice;
        this.assignedGift = assignedGift;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public ChildGender getGender() {
        return gender;
    }

    public boolean hasBeenNice() {
        return hasBeenNice;
    }

    public Gift getAssignedGift() {
        return assignedGift;
    }
}
